tests/unit/customizations/s3/test_s3handler.py

```diff
@@ -903,190 +904,124 @@ def test_remove_bucket(self):
-                self.assertTrue(mock_set_size_from_s3.called)
-                # Ensure that this download would have been a multipart
-                # download.
-                self.assertTrue(mock_enqueue_range_tasks.called)
-
-    def test_enqueue_range_download_tasks_stream(self):
-        s3handler = S3StreamHandler(
-            self.session, self.params,
-            runtime_config=runtime_config(multipart_chunksize=100))
-        s3handler.executor = mock.Mock()
-        fileinfo = FileInfo('filename', operation_name='download',
-                            is_stream=True, size=100)
-        s3handler._enqueue_range_download_tasks(fileinfo)
-        # Ensure that no request was sent to make a file locally.
-        submitted_tasks = s3handler.executor.submit.call_args_list
-        self.assertNotEqual(type(submitted_tasks[0][0][0]),
-                            CreateLocalFileTask)
+        expected_size = 6
+        self.params['expected_size'] = expected_size
+        handler = S3TransferStreamHandler(self.session, self.params)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            handler.call([file], self.transfer_manager)
+
+        self.assertEqual(self.transfer_future.meta.size, expected_size)
+
+    def test_upload_modifies_chunksize_if_too_low(self):
+        config = runtime_config(multipart_chunksize=1)
+        handler = S3TransferStreamHandler(
+            self.session, self.params, runtime_config=config)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            handler.call([file], self.transfer_manager)
+
+        self.assertTrue(
+            handler.config.multipart_chunksize >= 5 * (1024 ** 2))
+
+    def test_upload_modifies_chunksize_if_too_high(self):
+        config = runtime_config(multipart_chunksize=6 * (1024 ** 3))
+        handler = S3TransferStreamHandler(
+            self.session, self.params, runtime_config=config)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            handler.call([file], self.transfer_manager)
+
+        self.assertTrue(
+            handler.config.multipart_chunksize <= 5 * (1024 ** 3))
+
+    def test_upload_modifies_chunksize_for_max_parts_if_size_known(self):
+        expected_size = 6 * (1024 ** 3)
+        self.params['expected_size'] = expected_size
+        config = runtime_config(multipart_chunksize=1)
+        handler = S3TransferStreamHandler(
+            self.session, self.params, runtime_config=config)
+        file = FileInfo('-', 'foo-bucket/bar.txt', is_stream=True,
+                        operation_name='upload')
+
+        with MockStdIn(b'foobar'):
+            handler.call([file], self.transfer_manager)
+
+        self.assertTrue(
```

Comment:
Helper methods for asserting that the chunksize is greater than the minimum and less than the maximum would be great here so that is more explicit. That took me some time to figure out that is what you were doing.